src/NCalc/Expression.cs

```diff
@@ -291,49 +288,44 @@ public object Evaluate()
             if ((Options & EvaluateOptions.IterateParameters) == EvaluateOptions.IterateParameters)
             {
                 int size = -1;
-                ParametersBackup = new Dictionary<string, object>();
-                foreach (string key in Parameters.Keys)
-                {
-                    ParametersBackup.Add(key, Parameters[key]);
-                }
-
-                ParameterEnumerators = new Dictionary<string, IEnumerator>();
 
                 foreach (object parameter in Parameters.Values)
                 {
                     if (parameter is IEnumerable enumerable)
                     {
-                        int localsize = 0;
+                        int localSize = 0;
                         foreach (object o in enumerable)
                         {
-                            localsize++;
+                            localSize++;
                         }
 
                         if (size == -1)
                         {
-                            size = localsize;
+                            size = localSize;
                         }
-                        else if (localsize != size)
+                        else if (localSize != size)
                         {
                             throw new EvaluationException("When IterateParameters option is used, IEnumerable parameters must have the same number of items");
                         }
                     }
                 }
 
+                var parameterEnumerators = new Dictionary<string, IEnumerator>();
+
                 foreach (string key in Parameters.Keys)
                 {
                     if (Parameters[key] is IEnumerable parameter)
                     {
-                        ParameterEnumerators.Add(key, parameter.GetEnumerator());
+                        parameterEnumerators.Add(key, parameter.GetEnumerator());
                     }
                 }
 
                 var results = new List<object>();
                 for (int i = 0; i < size; i++)
                 {
-                    foreach (string key in ParameterEnumerators.Keys)
+                    foreach (string key in parameterEnumerators.Keys)

```

Comment:
Iterate over the dictionary, not the keys since we need to get the value anyway.  

`foreach (var parameterEnumerator in parameterEnumerators)`

Reply:
Looks like this change doesn't have any benefits, because we also need a key